include/mgba/internal/gba/flashrom.h

```diff
@@ -0,0 +1,71 @@
+#ifndef GBA_FLASHROM_H
+#define GBA_FLASHROM_H
+
+#include <mgba-util/common.h>
+#include <mgba-util/memory.h>
+
+CXX_GUARD_START
+
+enum FlashROMType {
+	FLASHROM_NONE,
+	FLASHROM_22XX,
+	FLASHROM_INTEL
+};

```

Comment:
This should probably be in the Savedata type enum instead. Actually, most of or all of this should be rolled into the savedata struct.

Reply:
I had a reason for separating it out. SMA4 needs 1M SRAM and will give warnings if 512K SRAM is present, but Pokemon Mystery Dungeon will corrupt its save if 1M SRAM is present and 512K SRAM is needed. Separating it out seemed reasonable.

In practice, I think there will only be 4 needed FlashROM configs for maximal compatibility (22xx 512K, 22xx 1M, 1M, Intel 512K, Intel 1M) so maybe it's better to just enumerate them in the normal save list.